File: examples-objc/LocusLabsSamples/LocusLabsSamples/SampleVCs/HeadlessMode.m
//
//  HeadlessMode.m
//  Copyright © 2020 LocusLabs. All rights reserved.
//

#import "HeadlessMode.h"
#import <LocusLabsSDK/LocusLabsSDK.h>

@interface HeadlessMode () <LLVenueDatabaseDelegate>

@property (nonatomic, strong) LLVenue           *venue;
@property (nonatomic, strong) LLVenueDatabase   *venueDatabase;

@end

@implementation HeadlessMode


#pragma mark Lifecycle

- (void)viewDidLoad {
    
    [super viewDidLoad];
    
    [LLLocusLabs setup].accountId = @"A11F4Y6SZRXH4X";
    
    // Get an instance of LLVenueDatabase, set it's mapview and register as its delegate
    self.venueDatabase = [LLVenueDatabase venueDatabase];
    self.venueDatabase.delegate = self;
    
    [self.venueDatabase loadVenue:@"lax"];
}

#pragma mark Delegates - LLVenueDatabase

- (void)venueDatabase:(LLVenueDatabase *)venueDatabase venueLoaded:(LLVenue *)venue {
    
    // Venue ready to use
    NSLog(@"Venue loaded and ready to use");
}

- (void)venueDatabase:(LLVenueDatabase *)venueDatabase venueLoadFailed:(NSString *)venueId code:(LLDownloaderError)errorCode message:(NSString *)message {

    // Handle failures here
}


@end

File: AtriusSamplesSwift/AtriusSamplesSwift/Samples/HeadlessMode.swift
//
//  HeadlessMode.swift
//  LocusLabsSamplesSwift
//
//  Created by Juan Kruger on 27/09/2022.
//  Copyright © 2022 LocusLabs. All rights reserved.
//


import UIKit

class HeadlessMode: UIViewController, LLVenueDatabaseDelegate {

    // Vars
    var venueDatabase:      LLVenueDatabase!
    var venue:              LLVenue?
    var mapView:            LLMapView?
    
    // MARK: Lifecycle
    override func viewDidLoad() {
        
        super.viewDidLoad()
        
        // Initialize the SDK with the accountId provided
        LLLocusLabs.setup().accountId = "A11F4Y6SZRXH4X"
        
        // Get an instance of LLVenueDatabase and register as its delegate
        venueDatabase = LLVenueDatabase()
        venueDatabase.delegate = self
                
        // Load the venue LAX async
        venueDatabase.loadVenue("lax")
    }
    
    // MARK: Delegates - LLVenueDatabase
    func venueDatabase(_ venueDatabase: LLVenueDatabase!, venueLoadCompleted venueId: String!) {
        
        // Venue ready to use
    }
    
    func venueDatabase(_ venueDatabase: LLVenueDatabase!, venueLoaded venue: LLVenue!) {
            
        self.venue = venue
    }
    
    func venueDatabase(_ venueDatabase: LLVenueDatabase!, venueLoadFailed venueId: String!, code errorCode: LLDownloaderError, message: String!) {
        
        // Handle failures here
    }
}


